### PR TITLE
Do not lick the supermatter, it is not made of candy.

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About The Pull Request
- Adds a special interaction for people attempting to lick the ~~candy~~supermatter.

## Why It's Good For The Game
It adds a little bit of extra fluff and humor.

## Changelog
:cl:
add: Nanotransen would like to make an official notice to all stations equipped with an SME (SuperMatter Engine) Since the last batch of crystals were shipped we have noticed an increased workplace accident occurrence rate in the engineering department. The supermatter is not made of candy. Do not lick it.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
Literally 1984.

**Comment by Tlaltecuhtli** [comment]
add emag effect to super matter

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -923,7 +923,38 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+		)
+		lick.Remove(user)
+		Consume(lick)
+		return
+
+	var/obj/item/bodypart/head/forehead = user.get_bodypart(BODY_ZONE_HEAD)
+	if(forehead)
+		user.visible_message(
+			"<span class='danger'>As [user]'s forehead bumps into [src], inducing a resonance... Everything goes silent before [user.p_their()] [forehead] flashes to ash!</span>",
+			"<span class='userdanger'>You feel your forhead bump into [src] and everything suddenly goes silent. As your head fills with ringing you come to realize that that was not a wise decision.</span>"
```
Comment (Iamgoofball) [inline code/modules/power/supermatter/supermatter.dm:948]: forehead

Comment (TemporalOroboros) [reply code/modules/power/supermatter/supermatter.dm:948]: Yes

Comment (Mothblocks) [reply code/modules/power/supermatter/supermatter.dm:948]: he meant theres a typo

File: code/modules/power/supermatter/supermatter.dm [hunk 2]
```diff
@@ -923,7 +923,38 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 	. = ..()
 	if(.)
 		return
-	dust_mob(user, cause = "hand")
+	if(user.incorporeal_move || user.status_flags & GODMODE)
+		return
+
+	. = TRUE
+	if(user.combat_mode || (user.zone_selected != BODY_ZONE_PRECISE_MOUTH))
```
Comment (Qustinnus) [inline code/modules/power/supermatter/supermatter.dm:930]: does this /really/ need to check combat mode?

File: code/modules/power/supermatter/supermatter.dm [hunk 3]
```diff
@@ -923,7 +923,38 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
+		dust_mob(user, cause = "hand")
+		return
+
+	var/obj/item/organ/tongue/lick = user.getorganslot(ORGAN_SLOT_TONGUE)
```
Comment (cacogen) [inline code/modules/power/supermatter/supermatter.dm:934]: I would name this licking_tongue

File: code/modules/power/supermatter/supermatter.dm (hunk 1 above)
Comment (TemporalOroboros) [reply code/modules/power/supermatter/supermatter.dm:948]: Yes. I'll fix it in a bit.

Comment (TemporalOroboros) [reply code/modules/power/supermatter/supermatter.dm:948]: Should be fixed in the next push.

File: code/modules/power/supermatter/supermatter.dm (hunk 3 above)
Comment (TemporalOroboros) [reply code/modules/power/supermatter/supermatter.dm:934]: Done. Should be in the next push.

File: code/modules/power/supermatter/supermatter.dm (hunk 2 above)
Comment (TemporalOroboros) [reply code/modules/power/supermatter/supermatter.dm:930]: It's probably not strictly necessary, but I might as well.

**Comment by Megamaw** [comment]
man, the fact it doesn't even tell you what it tastes like after you lick it and die is _**heartbreaking.**_

good pr, though.

**Review by Cyberboss — DISMISSED** [review]
~~I'm gonna veto the SM dusting individual bodyparts. It's not lore-friendly.~~

File: code/modules/power/supermatter/supermatter.dm (hunk 2 above)
Comment (Cyberboss) [reply code/modules/power/supermatter/supermatter.dm:930]: Let's not, please

Comment (Qustinnus) [reply code/modules/power/supermatter/supermatter.dm:930]: No, we should be limiting the amount of combat mode checks.

Comment (Qustinnus) [reply code/modules/power/supermatter/supermatter.dm:930]: (and be removing any that are left so we don't require modal interactions)

**Comment by TemporalOroboros** [comment]
> 
> 
> I'm gonna veto the SM dusting individual bodyparts. It's not lore-friendly.

The supermatter already dusts individual bodyparts. Specifically arms and brains.

**Pull request merged**